packages/mobile/src/utils/reachability.ts

```diff
@@ -0,0 +1,81 @@
+import { reachabilityActions } from '@audius/common'
+import type { NetInfoState } from '@react-native-community/netinfo'
+import NetInfo from '@react-native-community/netinfo'
+import { debounce } from 'lodash'
+import { AppState } from 'react-native'
+
+import { env } from 'app/services/env'
+import { dispatch } from 'app/store'
+
+const REACHABILITY_URL = env.REACHABILITY_URL
+
+// Latest connectivity value
+export const Connectivity: { netInfo: NetInfoState | null } = { netInfo: null }
+
+export const checkNetInfoReachability = (netInfo: NetInfoState | null) => {
+  if (!netInfo) return true
+
+  const { isInternetReachable } = netInfo
+  return !!isInternetReachable
+}
+
+// Check that a response from REACHABILITY_URL is valid
+const isResponseValid = (response: Response | undefined) =>
+  response && response.ok
+
+const pingTest = async () => {
+  // If there's no reachability url available, consider ourselves reachable
+  if (!REACHABILITY_URL) {
+    console.warn('No reachability url provided')
+    return true
+  }
+
+  try {
+    const response = await fetch(REACHABILITY_URL, { method: 'GET' })
+
+    if (isResponseValid(response)) {
+      console.debug('Reachability call succeeded')
+      return true
+    }
+    console.debug('Reachability call failed')
+    return false
+  } catch {
+    console.debug('Reachability call failed')
+    return false
+  }
+}
+
+const updateReachability = async (netInfoState: NetInfoState) => {
+  Connectivity.netInfo = netInfoState
+
+  const newValue = checkNetInfoReachability(netInfoState)
+  if (!newValue) {
+    // Don't trust offline signal while app is not in focus
+    // We check on refocus
+    const appState = AppState.currentState
+    if (appState !== 'active') return
+
+    // Perform our own reachability test to be extra sure we're offline
+    const reachable = await pingTest()
```

Comment:
added ping test
moved app state test to only block disconnecting events

Reply:
def supportive of this ping test now in addition to the netinfo stuff. ideally we can move this to a route on discovery node so we don't depend on audius.co from here, but that can be left for another day